File: src/main/java/co/uk/isxander/lunarspoof/websocket/auth/packet/LunarPacket.java
package co.uk.isxander.lunarspoof.websocket.auth.packet;

import co.uk.isxander.lunarspoof.websocket.auth.LunarAuthWebSocket;
import co.uk.isxander.xanderlib.utils.json.BetterJsonObject;

public abstract class LunarPacket {

    public abstract String getName();
    public abstract void process(LunarAuthWebSocket ws);

    public void processJson(BetterJsonObject json) {
    }

}

File: src/main/kotlin/dev/isxander/lunarspoof/websocket/auth/packet/LunarPacket.kt
package dev.isxander.lunarspoof.websocket.auth.packet

import dev.isxander.lunarspoof.websocket.auth.LunarAuthWebSocket
import gg.essential.api.utils.JsonHolder

abstract class LunarPacket {
    abstract val name: String
    abstract fun process(ws: LunarAuthWebSocket)

    open fun processJson(json: JsonHolder) {}
}